src/transformers/models/superpoint/image_processing_superpoint.py

```diff
@@ -270,3 +277,54 @@ def preprocess(
         data = {"pixel_values": images}
 
         return BatchFeature(data=data, tensor_type=return_tensors)
+
+    def post_process_keypoint_detection(
+        self, outputs: "SuperPointKeypointDescriptionOutput", target_sizes: Union[TensorType, List[Tuple]]
+    ):
+        """
+        Converts the raw output of [`SuperPointForKeypointDetection`] into lists of keypoints, scores and descriptors
+        with coordinates absolute to the original image sizes.
+
+        Args:
+            outputs ([`SuperPointKeypointDescriptionOutput`]):
+                Raw outputs of the model.
+            target_sizes (`torch.Tensor` or `List[Tuple[int, int]]`, *optional*):
+                Tensor of shape `(batch_size, 2)` or list of tuples (`Tuple[int, int]`) containing the target size
+                `(height, width)` of each image in the batch. This must be the original
+                image size (before any processing).
+        Returns:
+            `List[Dict]`: A list of dictionaries, each dictionary containing the keypoints, scores and descriptors for
+            an image in the batch as predicted by the model.
+        """
+        if len(outputs.mask) != len(target_sizes):
+            raise ValueError("Make sure that you pass in as many target sizes as the batch dimension of the mask")
+
+        if isinstance(target_sizes, List):
+            image_sizes = torch.tensor(target_sizes)
+        else:
+            if target_sizes.shape[1] != 2:
+                raise ValueError(
+                    "Each element of target_sizes must contain the size (h, w) of each image of the batch"
+                )
+            image_sizes = target_sizes
+
+        masked_keypoints = outputs.keypoints.clone()
+
+        for keypoints, image_size in zip(masked_keypoints, image_sizes):
+            keypoints[:, 0] = keypoints[:, 0] * image_size[1]
+            keypoints[:, 1] = keypoints[:, 1] * image_size[0]
```

Comment:
I think? Otherwise we're just taking the first and second elements of the batch. 

If I'm right - I'm surprised this wasn't caught. Has this been tested on batch inputs? 

```suggestion
            keypoints[:, 0] = keypoints[:, 0] * image_size[:, 1]
            keypoints[:, 1] = keypoints[:, 1] * image_size[:, 0]
```

Reply:
I don't think there is a problem with this part. After checking, `image_size` is indeed a `(2,)` tensor, `keypoints` as `(n, 2)`, so `keypoints[:, 0] = keypoints [:, 0] * image_size[1]` multiplies all elements x elements from `keypoints` with the scalar in `image_size`